lib/game.rb

```diff
@@ -0,0 +1,71 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2, :token
+
+  WIN_COMBINATIONS = [
+  [0,1,2], #Top row
+  [3,4,5], #Middle row
+  [6,7,8], #Bottom row
+  [0,3,6], #Left column
+  [1,4,7], #Middle column
+  [2,5,8], #Right column
+  [0,4,8], #Left start diagonal
+  [2,4,6]  #Right start diagonal
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    if board.turn_count % 2 == 0
+      player_1
+    elsif board.turn_count % 2 == 1
+      player_2
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.detect do |combination|
+      board.cells[combination[0]] == board.cells[combination[1]] &&
+      board.cells[combination[0]] == board.cells[combination[2]] && board.cells[combination[0]] != " "
+    end
+  end
+
+  def draw?
+    board.full? && !won?
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def winner
+    if won?
+      board.cells[won?[0]]
+    end
+  end
+
+  def turn
+    input = current_player.move(@board)
+    if board.valid_move?(input)
+      board.update(input, current_player)
+    else
+      turn
+    end
+  end
+
+  def play
+    until over?
+      turn
+    end
+    if won?
+      puts "Congratulations #{winner}!"
+    elsif draw?
+      puts "Cat's Game!"
+    end
+  end
+
+end
```

Comment:
Looks like there's no executable to actually run this game.  Right now, although the tests can use your code, how would you actually run this from the terminal?  Create an executable file in your bin folder, something like `clitictactoe`, and it would only take a few lines of code to get it going:
```
 #!/usr/bin/env ruby
 require_relative '../config/environment'
CLI.new.call
```
Where `CLI` is a new class responsible for input and output from a user.  Its responsibilities would include greeting a player, determine how many players and instantiate a new game, (once started, the game play would yield to the Game class), asking a user if another game is desired, and saying 'goodbye' when a user decides not to play again.